### DISCUSSION TITLE
Cable bulk import error when importing cables with circuit terminations

### DISCUSSION THREAD

**Discussion by KPTheProf** [netbox-community/netbox#16411] [Help Wanted!]
### Deployment Type

Self-hosted

### NetBox Version

v4.0.3

### Python Version

3.12

### Steps to Reproduce

When trying to use the bulk cable import feature to import cables with circuit terminations, I'm getting the error "Record 1 side_b_device: This field is required".

Circuit terminations doen't have a device, but the side_?_device field is mandatory. What should I be putting in the field, if it can't be empty?

To reproduce:
I have an existing device BT-NTU-01 with a port Network 1.
I have an existing circuit, CIRCUIT-01 and I'm trying to connect to Termination Z

```
side_a_type,side_a_device,side_a_name,side_b_type,side_b_device,side_b_name,type,status
dcim.interface,BT-NTU-01,Network 1,circuits.circuittermination,,CIRCUIT-01: Termination Z,smf,connected
```

### Expected Behavior

Should be able to import cables connected to circuit terminations.

### Observed Behavior

![image](https://github.com/netbox-community/netbox/assets/56978198/513587cb-6729-4afb-829f-c8fe425fcc41)

**Comment by jeffgdotorg** [comment]
This issue sounds more like a request for guidance than a bug report, so I'm converting it to a discussion in the _Help Wanted_ category. If the conversation there leads back to bug territory, then we'll meet back here in Issues 😉

**Reply by jjoves73** [reply]
I believe what he is saying is that - there is a termination type called "circuits.circuittermination" - however there is no other field we can use that relates to a "circuit" - If we are going to connect a port to a circuit.

dcim.interface, rearport, frontport, etc has the fields for the 'device' and 'name'
circuit termination usually relates to a Provider and CID - which is not included in the Header/Field options list

**Reply by pheus** [reply]
Thanks for raising this! You're on the right track. From the current implementation, the cable CSV importer is built around **device** components. You choose each termination via `side_X_type` and `side_X_name` (Interface, PowerPort, ConsolePort, etc.), but each side also requires a `side_X_device`. Circuit terminations live in the **Circuits** app, not under **Devices**, so they don’t have a `device` field - only a circuit ID and a termination side (A/Z). That’s why we’re seeing “side_b_device: This field is required”.

Hope that clarifies the behavior. If CSV support for circuit terminations would be helpful, it might be worth opening the feature request, again - or using the UI/API as a workaround in the meantime.

**Reply by jjoves73** [reply]
So at the moment - the circuits.circuittermination field is useless for now since we can’t select that, 
If I select that as termination type on a or b…it won’t have enough information

**Comment by BahaeddinSagar** [comment]
did you find a solution for this ?
